tensorflow/python/ops/check_ops.py

```diff
@@ -1504,6 +1505,269 @@ def assert_type(tensor, tf_type, message=None, name=None):
     return control_flow_ops.no_op('statically_determined_correct_type')
 
 
+def _dimension_sizes(x):
+  """Gets the dimension sizes of a tensor `x`.
+
+  If a size can be determined statically it is returned as an integer,
+  otherwise as a tensor.
+
+  If `x` is a scalar it is treated as rank 1 size 1.
+
+  Args:
+    x: A `Tensor`.
+
+  Returns:
+    Dimension sizes.
+  """
+  dynamic_shape = array_ops.shape(x)
+  rank = x.get_shape().rank
+  rank_is_known = rank is not None
+  if rank_is_known and rank == 0:
+    return tuple([1])
+  if rank_is_known and rank > 0:
+    static_shape = x.get_shape().as_list()
+    sizes = [
+        int(size) if size is not None else
+        dynamic_shape[i]
+        for i, size in enumerate(static_shape)
+    ]
+    return sizes
+  has_rank_zero = math_ops.equal(array_ops.rank(x), 0)
+  return control_flow_ops.cond(
+      has_rank_zero,
+      lambda: array_ops.constant([1]),
+      lambda: dynamic_shape
+  )
+
+
+def _symbolic_dimension_sizes(symbolic_shape):
+  if len(symbolic_shape) == 0:
+    return tuple([1])
+  return symbolic_shape
+
+
+def _has_known_value(dimension_size):
+  not_none = dimension_size is not None
+  try:
+    int(dimension_size)
+    can_be_parsed_as_int = True
+  except (ValueError, TypeError):
+    can_be_parsed_as_int = False
+  return not_none and can_be_parsed_as_int
+
+
+def _is_symbol_for_any_size(symbol):
+  return symbol in [None, '.']
+
+
+def _is_symbol_for_unspecified_dims(symbol):
+  return symbol in [..., '*']
```

Comment:
I think the ellipsis here is causing some of our python2 tests to fail with 

  File "tensorflow/python/ops/check_ops.py", line 1561
    return symbol in [..., '*']
SyntaxError: Type annotations and ellipsis literal not enabled.

Reply:
I see. Apparently python2 does not support the Ellipsis literal, but it does have the Ellipsis type. So by replacing '...' by 'Ellipsis' in the symbol list it will be made backwards compatible with python2. I can push that change

Reply:
Yes, thanks!

Reply:
Pushed